### PR TITLE
refactor: move internal/pkg/version to internal/version

### PR DESCRIPTION
**Pull request by mwbrooks**
### Changelog

- N/A

### Summary

This pull request refactors `internal/pkg/version` to `internal/version`. The motivation is to reduce the usage of `internal/pkg` - all of these packages should eventually find a home outside of `internal/pkg`.

While this is a small change, it does touch our build configuration.

### Test Steps

```bash
$ lack --version
# → Confirm the correct SHA is in the version: `-g<SHA>`
#   → "v3.15.0-26-gc082ce3"
```

### Requirements

- [x] I've read and understood the [Contributing Guidelines](https://github.com/slackapi/slack-cli/blob/main/.github/CONTRIBUTING.md) and have done my best effort to follow them.
- [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).

### CODE REVIEW

File: internal/version/version.go
Comment (mwbrooks) [inline internal/version/version.go:1]: farewell: So long and thanks for all the fish pkg 🐬

File: internal/shared/clients.go [hunk 1]
```diff
@@ -95,8 +95,7 @@ func NewClientFactory(options ...func(*ClientFactory)) *ClientFactory {
 	// TODO: Temporary hack to get around circular dependency in internal/api/client.go since that imports version
```
Comment (mwbrooks) [inline internal/shared/clients.go:95]: thought: I did a little research and I believe this TODO is was solved by our `slackcontext.Version(ctx)` improvements. 

I don't want to remove this comment in this PR, because it's out-of-scope, but I'll start to work on a follow-up PR that removes this comment and potentially some other cleanup related to this.

Comment (zimeg) [reply internal/shared/clients.go:95]: @mwbrooks Do be warned - The versions expected for `login` methods upstream can be reliant on something related to version prefixes... 👻 

I'm optimistic from the `slackcontext` enhancements as well but wanted to share prior edge known.

File: cmd/doctor/check.go [hunk 2]
```diff
@@ -26,12 +26,12 @@ import (
 	"github.com/slackapi/slack-cli/internal/config"
 	"github.com/slackapi/slack-cli/internal/deputil"
 	"github.com/slackapi/slack-cli/internal/iostreams"
-	"github.com/slackapi/slack-cli/internal/pkg/version"
 	"github.com/slackapi/slack-cli/internal/shared"
 	"github.com/slackapi/slack-cli/internal/shared/types"
 	"github.com/slackapi/slack-cli/internal/slackerror"
 	"github.com/slackapi/slack-cli/internal/style"
 	"github.com/slackapi/slack-cli/internal/update"
+	"github.com/slackapi/slack-cli/internal/version"
```
Comment (zimeg) [inline cmd/doctor/check.go:34]: 🔭 praise: This is a nice `diff` to find!

File: internal/shared/clients.go (hunk 1 above)
Comment (mwbrooks) [reply internal/shared/clients.go:95]: 🧠 Oh yes, that's a good point! I've put together PR #429 and I'll circle back to test the login. I could definitely use your help on thinking about edge-cases in it.
